File: dataset/output/OPENAI_gpt-4o/Task126_OPENAI_gpt-4o.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

void str_tolower(char* str) {
    for (int i = 0; str[i]; i++) {
        str[i] = tolower(str[i]);
    }
}

int main() {
    char input[50];

    printf("Enter user role (admin/user/guest): ");
    fgets(input, 50, stdin);

    // Remove newline character if it exists
    input[strcspn(input, "
")] = 0;

    str_tolower(input);
    
    if (strcmp(input, "admin") == 0) {
        printf("Redirecting to admin page...
");
    } else if (strcmp(input, "user") == 0) {
        printf("Redirecting to user page...
");
    } else if (strcmp(input, "guest") == 0) {
        printf("Redirecting to guest page...
");
    } else {
        printf("Invalid role!
");
    }

    return 0;
}

File: dataset/output/OPENAI_gpt-4o/Task126_OPENAI_gpt-4o.java
import java.util.Scanner;

public class RedirectUser {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter user role (admin/user/guest): ");
        String input = scanner.nextLine().trim().toLowerCase();

        switch (input) {
            case "admin":
                System.out.println("Redirecting to admin page...");
                break;
            case "user":
                System.out.println("Redirecting to user page...");
                break;
            case "guest":
                System.out.println("Redirecting to guest page...");
                break;
            default:
                System.out.println("Invalid role!");
                break;
        }
        scanner.close();
    }
}